ucsm_apis/admin/tacacsplus.py

```diff
@@ -0,0 +1,437 @@
+        UcsOperationError: If AaaTacacsPlusProvider is not present
+
+    Example:
+        tacacsplus_provider_modify(handle, "test_tacac_prov", timeout="5")
+    """
+
+    mo = tacacsplus_provider_get(handle, name,
+                                 caller="tacacsplus_provider_modify")
+    mo.set_prop_multiple(**kwargs)
+    handle.set_mo(mo)
+    handle.commit()
+    return mo
+
+
+def tacacsplus_provider_delete(handle, name):
+    """
+    deletes a tacacsplus provider
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of tacacsplus provider
+
+    Returns:
+        None
+
+    Raises:
+        UcsOperationError: If AaaTacacsPlusProvider is not present
+
+    Example:
+        tacacsplus_provider_delete(handle, name="test_tacac_prov")
+    """
+
+    mo = tacacsplus_provider_get(handle, name,
+                                 caller="tacacsplus_provider_delete")
+    handle.remove_mo(mo)
+    handle.commit()
+
+
+def tacacsplus_provider_group_create(handle, name, descr=None, **kwargs):
+    """
+    Creates a tacacsplus provider group
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of tacacsplus provider group
+        descr (string): descr
+        **kwargs: Any additional key-value pair of managed object(MO)'s
+                  property and value, which are not part of regular args.
+                  This should be used for future version compatibility.
+    Returns:
+        AaaTacacsPlusProvider: Managed Object
+
+    Example:
+        tacacsplus_provider_create(handle, name="test_prov_grp")
+    """
+    from ucsmsdk.mometa.aaa.AaaProviderGroup import AaaProviderGroup
+
+    mo = AaaProviderGroup(parent_mo_or_dn=_base_dn, name=name, descr=descr)
+    mo.set_prop_multiple(**kwargs)
+    handle.add_mo(mo, True)
+    handle.commit()
+    return mo
+
+
+def tacacsplus_provider_group_get(handle, name,
+                                  caller="tacacsplus_provider_group_get"):
+    """
+    Gets tacacsplus provider group
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of tacacsplus provider group
+
+    Returns:
+        AaaTacacsPlusProvider: Managed Object OR None
+
+    Example:
+        tacacsplus_provider_group_get(handle, name="test_prov_grp")
+    """
+
+    dn = _base_dn + "/providergroup-" + name
+    mo = handle.query_dn(dn)
+    if mo is None:
+        raise UcsOperationError(caller,
+                        "Tacacsplus  Provider Group '%s' does not exist" % dn)
+    return mo
+
+
+def tacacsplus_provider_group_exists(handle, name, **kwargs):
+    """
+    checks if a tacacsplus provider group exists
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of tacacsplus provider group
+        **kwargs: key-value pair of managed object(MO) property and value, Use
+                  'print(ucscoreutils.get_meta_info(<classid>).config_props)'
+                  to get all configurable properties of class
+
+    Returns:
+        (True/False, MO/None)
```

Comment:
should we say type of the MO instead of MO ?
for e.g. 
`(True/False, AaaUser/None)`